src/components/Sidebar/Tabs/FilterScreen.tsx

```diff
@@ -136,19 +178,41 @@ function QuantitativeFilters(props: FilterGroupProps) {
           ))}
         </select>
       </label>
+      {ranges.map((r, i) => (
+        <div style={{ display: 'flex' }}>
+          <RangeSlider
+            width={300}
+            domain={bounds}
+            values={r}
+            onUpdate={(update) => updateRange(update, i)}
+          />
+          <button
+            className="close-slider wrapper"
+            onClick={() => deleteRange(i)}
+          >
+            <X size={20} />
+          </button>
+        </div>
+      ))}
+      <button
+        className="wrapper block add-range"
+        onClick={() => updateRange(bounds, ranges.length)}
+      >
+        + <Sliders />
+      </button>
     </div>
   );
 }
 
 function CategoricalFilters(props: FilterGroupProps) {
-  const [graphData] = useModelState<GraphData<string>>('graph_data');
-  const [graphSpec, setGraphSpec] = useModelState<GraphSpec>('graph_spec');
+  const [graphData] = useModelState('graph_data');
+  const [graphSpec, setGraphSpec] = useModelState('graph_spec');
   const categories = useMemo(getCategories, []);
```

Comment:
Just have a question about useMemo(). What is the benefit of using useMemo instead of useState as here we don't depend on anything?